compiler/test/graindoc/types.expected.md

```diff
@@ -0,0 +1,57 @@
+---
+title: TypeGrainDoc
+---
+
+## Types
+
+Type declarations included in the TypeGrainDoc module.
+
+### TypeGrainDoc.**R**
+
+```grain
+record R {
+  x: Number,
+  y: String,
+}
+```
+
+A record
+
+Fields:
+
+|name|type|description|
+|----|----|-----------|
+|`x`|`Number`|Record field x|
+|`y`|`String`|Record field y<br />Second line|
+
+### TypeGrainDoc.**E**
+
+```grain
+enum E {
+  Variant1,
+  Variant2(String, Number),
+  Variant3{
+    a: Number,
+    f: Number => Number,
+  },
+}
+```
+
+An enum
+
+Variants:
+
+|name|data|description|
+|----|----|-----------|
+|`Variant1`||Enum variant|
+|`Variant2`|`(String, Number)`|Another variant|
+|`Variant3`|<table><thead><tr><th>name</th><th>type</th><th>description</th></tr></thead><tbody><tr><td>`a`</td><td>`Number`</td><td>Record field a</td></tr><tr><td>`f`</td><td>`Number => Number`</td><td>Function</td></tr></tbody></table><br /><br />|Record variant|
```

Comment:
Sorry for the delay here. I don't really like this layout. How about we do this?

````suggestion
Variants:

```grain
Variant1
```
Enum variant

```grain
Variant2(String, Number)
```
Another variant

```grain
Variant3{
  a: Number,
  f: Number => Number,
}
```
Record Variant

Fields:

|name|type|description|
|----|----|-----------|
|`a`|`Number`|Record field a|
|`f`|`Number => Number`|Function|
````

I was also thinking about making these a bulleted list, but I believe that'd be hard while also making them grain-syntax blocks.

Reply:
Assuming that record fields are being kept in a table I'm not a huge fan of having enum variants not be in a table. What about something like this?
```
|variant|description|
|------------|-----------|
|`Variant1`|Enum variant|
|`Variant2(String, Number)`|Another variant|
|<code>Variant3{<br>&nbsp;&nbsp;a: Number,<br>&nbsp;&nbsp;f: Number => Number<br>}</code>|Record variant<br><ul><li>`a`: Record field a</li><li>`f`: Function</li></ul>|
```

![image](https://github.com/grain-lang/grain/assets/48104782/62943bca-8d5f-4a16-aa6f-ef93e676668c)

Reply:
I don't like them in a table. It's too dense

Reply:
Applied your suggested format